### PR TITLE
Supporting multiple SDFG source files

### PR DESCRIPTION
**Pull request by benibenj**
- Supporting multiple SDFG source files (breakpoints will be mapped from all sourcefiles, all sourcefiles get the jump to C++/SDFG, ...)
- When using the cache config 'hash' a message pops up indicating restricted features supported due to there not being any mapping
- When the SDFG is created with the API we don't map the python breakpoints
- Some code clean up and bug fixes

### CODE REVIEW

**Review by tbennun — APPROVED** [review]
Looks good to me. @phschaad any comments?

File: src/debugger/breakpointHandler.ts [hunk 1]
```diff
@@ -166,83 +178,101 @@ export class BreakpointHandler extends vscode.Disposable {
...
+        this.setAllBreakpoints();
+        vscode.debug.activeDebugSession?.customRequest("continue");
+        this.showMenu(true);
         this.saveState();
     }
 
     public setAllBreakpoints() {
         // Map and set all Breakpoints set in the dace (python) code
         vscode.debug.breakpoints.filter(pyFilter).forEach(bp => {
+            console.log(bp);
```
Comment (phschaad) [inline src/debugger/breakpointHandler.ts:275]: Necessary?

File: src/debugger/daceListener.ts [hunk 2]
```diff
@@ -51,6 +56,18 @@ export class DaceListener extends vscode.Disposable{
             case "registerFunction":
                 BreakpointHandler.getInstance()?.registerFunction(data);
                 break;
+            case "restrictedFeatures":
+                if (!this.hasIndicatedRestricted) {
+                    if (data.reason === 'config.cache.hash') {
+                        // When using the cache config 'hash' the mapping won't
+                        // be created and so not all features can be used
+                        const msg = "Due to using the cache configuration " +
```
Comment (phschaad) [inline src/debugger/daceListener.ts:64]:
```suggestion
                        const msg = "Due to the use of the cache configuration " +
```

**Review by phschaad — APPROVED** [review]
LGTM too, thanks!

File: src/debugger/breakpointHandler.ts (hunk 1 above)
Comment (tbennun) [reply src/debugger/breakpointHandler.ts]: I had the same question :)

Comment (benibenj) [reply src/debugger/breakpointHandler.ts]: no, removed it

**Pull request merged**